roles/openshift_facts/library/openshift_facts.py

```diff
@@ -1163,6 +1167,49 @@ def init_local_facts(self, facts=None):
         self.changed = changed
         return new_local_facts
 
+    def init_in_docker_facts(self, facts):
+        facts['is_containerized'] = os.path.isfile('/run/ostree-booted')
+        deployment_type = facts['common']['deployment_type']
+
+        # TODO: Most of these shouldn't actually make it into the facts
+        docker = dict()
+        if deployment_type in ['enterprise','openshift-enterprise']:
+            docker['master_image_name'] = 'openshift3/ose'
+            docker['ovs_image_name'] = 'openshift3/openvswitch'
+            docker['node_image_name'] = 'openshift3/node'
+            docker['etcd_image_name'] = 'rhel7/etcd'
+        elif deployment_type == 'atomic-enterprise':
+            docker['master_image_name'] = 'aep3/aep'
+            docker['ovs_image_name'] = 'aep3/openvswitch'
+            docker['node_image_name'] = 'aep3/node'
+            docker['etcd_image_name'] = 'rhel7/etcd'
+        else:
+            docker['master_image_name'] = 'openshift/origin'
+            docker['ovs_image_name'] = 'openshift/openvswitch'
+            docker['node_image_name'] = 'openshift/node'
+            docker['etcd_image_name'] = 'rhel7/etcd'
+
+        docker['image_version'] = 'latest'
+        docker['etcd_image_version'] = 'latest'
+
+        docker['master_image'] = "%s:%s" % (docker['master_image_name'], docker['image_version'])
+        docker['ovs_image'] = "%s:%s" % (docker['ovs_image_name'], docker['image_version'])
+        docker['node_image'] = "%s:%s" % (docker['node_image_name'], docker['image_version'])
+        docker['etcd_image'] = "%s:%s" % (docker['etcd_image_name'], docker['etcd_image_version'])
+
+        # shared /tmp/openshift vol is for file exchange with ansible
+        # --privileged is required to read the config dir
+        # --net host to access openshift from the container
+        # maybe -v /var/run/docker.sock:/var/run/docker.sock is required as well
+        docker['runner'] = "docker run --rm --privileged --net host -v /tmp/openshift:/tmp/openshift -v {datadir}:{datadir} -v {confdir}:{confdir} -e KUBECONFIG={confdir}/master/admin.kubeconfig {image}".format(confdir=facts['common']['config_base'], datadir=facts['common']['data_dir'], image=docker['master_image'])
+
+        if facts['is_containerized']:
+            facts['common']['client_binary'] = '%s cli' % docker['runner']
+            facts['common']['admin_binary'] = '%s admin' % docker['runner']
+
```

Comment:
Thoughts on the openshift_cli role I've added? in particular here's the wrapper https://github.com/openshift/openshift-ansible/pull/808/files#diff-59736f30a3d4d0e42b839c6856a5905a